lib/modules/datasource/index.ts

```diff
@@ -139,20 +139,21 @@ async function mergeRegistries(
   for (const registryUrl of registryUrls) {
     try {
       const res = await getRegistryReleases(datasource, config, registryUrl);
-      if (res) {
-        if (combinedRes) {
-          for (const existingRelease of combinedRes.releases || []) {
-            existingRelease.registryUrl = combinedRes.registryUrl;
```

Comment:
```diff
          for (const existingRelease of combinedRes.releases || []) {
-           existingRelease.registryUrl = combinedRes.registryUrl;
+           existingRelease.registryUrl ??= combinedRes.registryUrl; 
          }

```

although an edge case, but this I think this will cause problems when more than 2 registry urls are set.
it will basically erase this prop as `combinedRes.registryUrl` is undefined after the third res response.